Testnet/Ticketbooth/Ticketbooth/TicketContract_1_0_0.cs

```diff
@@ -0,0 +1,456 @@
+        Assert(seats.Length <= MAX_SEATS, $"Cannot handle more than {MAX_SEATS} seats");
+        var tickets = new Ticket[seats.Length];
+
+        for (int i = 0; i < seats.Length; i++)
+        {
+            tickets[i] = new Ticket { Seat = seats[i] };
+        }
+
+        Log(new Venue { Name = venueName });
+        Owner = Message.Sender;
+        Tickets = tickets;
+    }
+
+    /// <summary>
+    /// Stores ticket data for the contract
+    /// </summary>
+    public Ticket[] Tickets
+    {
+        get => PersistentState.GetArray<Ticket>(nameof(Tickets));
+        private set => PersistentState.SetArray(nameof(Tickets), value);
+    }
+
+    private ulong EndOfSale
+    {
+        get => PersistentState.GetUInt64(nameof(EndOfSale));
+        set => PersistentState.SetUInt64(nameof(EndOfSale), value);
+    }
+
+    private ulong ReleaseFee
+    {
+        get => PersistentState.GetUInt64(nameof(ReleaseFee));
+        set
+        {
+            PersistentState.SetUInt64(nameof(ReleaseFee), value);
+            Log(new TicketReleaseFee { Amount = value });
+        }
+    }
+
+    private ulong NoRefundBlockCount
+    {
+        get => PersistentState.GetUInt64(nameof(NoRefundBlockCount));
+        set
+        {
+            PersistentState.SetUInt64(nameof(NoRefundBlockCount), value);
+            Log(new NoRefundBlocks { Count = value });
+        }
+    }
+
+    private bool RequireIdentityVerification
+    {
+        get => PersistentState.GetBool(nameof(RequireIdentityVerification));
+        set
+        {
+            PersistentState.SetBool(nameof(RequireIdentityVerification), value);
+            Log(new IdentityVerificationPolicy { RequireIdentityVerification = value });
+        }
+    }
+
+    private Address Owner
+    {
+        get => PersistentState.GetAddress(nameof(Owner));
+        set => PersistentState.SetAddress(nameof(Owner), value);
+    }
+
+    private bool SaleOpen
+    {
+        get
+        {
+            var endOfSale = EndOfSale;
+            return endOfSale != 0 && Block.Number < endOfSale;
+        }
+    }
+
+    /// <summary>
+    /// Starts a ticket sale, when no sale is running
+    /// </summary>
+    /// <param name="ticketsBytes">The serialized array of tickets</param>
+    /// <param name="showName">Name of the event or performance</param>
+    /// <param name="organiser">The organiser or artist</param>
+    /// <param name="time">Unix time for the event</param>
+    /// <param name="endOfSale">The block at which the sale ends</param>
+    public void BeginSale(byte[] ticketsBytes, string showName, string organiser, ulong time, ulong endOfSale)
+    {
+        Assert(Message.Sender == Owner, "Only contract owner can begin a sale");
+        Assert(EndOfSale == default(ulong), "Sale currently in progress");
+        Assert(Block.Number < endOfSale, "Sale must finish in the future");
+
+        var tickets = Serializer.ToArray<Ticket>(ticketsBytes);
```

Comment:
It looks like this is just used to set the ticket price. If so, you could consider just passing in an ordered array of prices to save a few bytes.

Reply:
Good idea! This will definitely save a good amount of gas, so I'll do this in the next version.